apps/src/blockly/addons/cdoVerticalFlyout.js

```diff
@@ -0,0 +1,87 @@
+import GoogleBlockly from 'blockly/core';
+
+export default class VerticalFlyout extends GoogleBlockly.VerticalFlyout {
+  /**
+   * @override
+   * This is copied almost entirely from
+   * https://github.com/google/blockly/blob/master/core/flyout_vertical.js#L322
+   * We override flyoutWidth to constrain it to be at most 40% of the total
+   * workspace space.
+   * Once https://github.com/google/blockly/issues/5684 is resolved, we can
+   * remove this duplicated code and use the API to set the max width of the flyout.
+   **/
+  reflowInternal_() {
+    this.workspace_.scale = this.getFlyoutScale();
+    var flyoutWidth = 0;
+    var blocks = this.workspace_.getTopBlocks(false);
+    for (let i = 0, block; (block = blocks[i]); i++) {
```

Comment:
I changed a couple things to make this pass our linter (mostly `let i` instead of `var i` in the loops, and `===` instead of `==`) but I think other than that, it makes sense to leave this in mostly ES5 style since it's an exact duplication of the existing google blockly code, but definitely open to thoughts if people disagree!